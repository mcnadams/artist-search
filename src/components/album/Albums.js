import React from 'react';
import PropTypes from 'prop-types';
import Album from './Album';
import { styles, stylesLI } from './css.js';

function Albums({ albums, artist }) {
  
  const albumList = albums.map((album, i) => {
    return (
      <li key={i} style={stylesLI}>
        <Album album={album} artist={artist} />
      </li>
    );
  });

  return (
    <ul style={styles}>
      {albumList}
    </ul>
  );
}

Albums.propTypes = {
  albums: PropTypes.array.isRequired,
  artist: PropTypes.string.isRequired
};

export default Albums;
