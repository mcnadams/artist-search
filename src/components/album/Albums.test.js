import React from 'react';
import { shallow } from 'enzyme';
import Albums from './Albums';

describe('Albums renders', () => {
  it('renders Albums', () => {
    const wrapper = shallow(<Albums artist={'lady'} albums={[{ date: 'yes', id: 'no' }]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
