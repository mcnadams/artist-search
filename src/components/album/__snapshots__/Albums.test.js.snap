// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Albums renders renders Albums 1`] = `
<ul
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexWrap": "wrap",
      "justifyContent": "space-around",
      "listStyle": "none",
    }
  }
>
  <li
    key="0"
    style={
      Object {
        "alignItems": "center",
        "background": "rgb(123, 247, 123)",
        "border": "solid",
        "display": "flex",
        "flexWrap": "wrap",
        "height": "300px",
        "justifyContent": "center",
        "margin": "5px",
        "textAlign": "center",
        "width": "225px",
      }
    }
  >
    <Album
      album={
        Object {
          "date": "yes",
          "id": "no",
        }
      }
      artist="lady"
    />
  </li>
</ul>
`;
